### DISCUSSION TITLE
Lf a proper way to Type Wrapped AsyncSelect on v5.2.1

### DISCUSSION THREAD

**Discussion by luiz504** [JedWatson/react-select#4905] [Q&A]
Hello :) 
I get stuck trying to wrap the asyncSelect default Props with CustomProps on V5;
Following below is a sample where I made a simple *Select* that works fine and an *AsyncSelect* with the same principle that doesn't work at all with custom props extending the default ones.
If someone could help I will be glad :P
https://codesandbox.io/s/react-select-v5-wrapping-qx2f0?file=/src/components/Select/index.tsx

**Comment by Methuselah96** [comment]
Did you forgot to extend `IProps` from `AsyncProps` in your example?

**Reply by luiz504** [reply]
My bad I was playing around after posting the link and forget to return the objection point. 
Btw https://imgur.com/a/X1TtCst those news AsyncProps arguments required that I`m having some trouble working with.

**Reply by binhxn** [reply]
I'm running into the exact same problem and it's still unclear how to go about this. The last thing I'd want to do is pass in `any` as arguments (i.e. `AsyncProps<any, any, any>`) -- which works.

**Reply by Methuselah96** [reply]
Take a look at the "Wrapping the Select component" example on the [TypeScript page](https://react-select.com/typescript#select-generics) of the docs. You should be able to follow the same basic pattern for `Async`.

**Reply by GustavoMelloGit** [reply]
I am not able to do that. Can you show me a example?

**Reply by GustavoMelloGit** [reply]
Just got it. If anyone needs it:
````ts
function InputAsyncSelectFormik<
  Option,
  IsMulti extends boolean = false,
  Group extends GroupBase<Option> = GroupBase<Option>,
>(
  props: AsyncProps<Option, IsMulti, Group> &
    RefAttributes<Select<Option, IsMulti, Group>>,
) {
  [...]
}
```

**Reply by chanm003** [reply]
@GustavoMelloGit Do you have a CodeSandBox?
 
For Select<> where is that being imported from?

Thanks!
